src/librustc_trans/mir/mod.rs

```diff
@@ -319,7 +319,9 @@ pub fn trans_mir<'a, 'tcx: 'a>(
     mircx.cleanup_kinds.iter_enumerated().map(|(bb, cleanup_kind)| {
         if let CleanupKind::Funclet = *cleanup_kind {
             let bcx = mircx.get_builder(bb);
-            bcx.set_personality_fn(mircx.ccx.eh_personality());
+            unsafe {
+                llvm::LLVMSetPersonalityFn(mircx.llfn, mircx.ccx.eh_personality());
+            }
```

Comment:
... am I right in inferring from this change that `bcx.llfn()` is not the same as `mircx.llfn`?  It seems surprising.

And If that is actually true... it probably could use a comment.